### ISSUE TITLE
Service can't access http://rancher-metadata

### ISSUE THREAD

**Issue by emcniece** [rawmind0/rancher-traefik#19]
I'm investigating a system where `rancher-traefik` is enabled and doesn't recognize any services and won't add any backends to the `.toml` files.

I have tested https://github.com/jmcarbo/rancher-traefik and it works, but that version doesn't have an admin UI or the SSL features that this library does.

On inspection, I noticed that both services start and print a log line that says that they are using `http://rancher-metadata` as the metadata URL. Running `docker exec -it [contID] curl http://rancher-metadata` on the `jmcarbo` service prints an expected response (some dates and a "latest" tag), but executing this command on the `rawmind0` service returns `curl: (6) Couldn't resolve host 'rancher-metadata'`.

I see that https://github.com/rawmind0/alpine-traefik/issues/3 deals with something that appeared to be similar, but ended up being a label issue on the Drupal service. I am currently only testing with 2 separate `whoami` services, and they work as expected with the `jmcarbo` version.

https://github.com/rancher/rancher/issues/5041 talks about working with this DNS search issue, and the final comment adds `dns_search: -rancher.internal` to the service. It seems that this is already present on the `rancher-traefik` service.

The logs for the service (monit, traefik, confd) don't seem to have anything abnormal in them. Any ideas for where to look? @rawmind0 I have granted access to your Github user at http://my-rancher.com:8080/ in case you wish to inspect further... many MANY thanks for your insight.

<img width="931" alt="screen shot 2016-10-30 at 3 04 11 pm" src="https://cloud.githubusercontent.com/assets/1182187/19840454/6eb235ec-9eb3-11e6-81a5-4be17eb2c20c.png">
<img width="1283" alt="screen shot 2016-10-30 at 3 03 46 pm" src="https://cloud.githubusercontent.com/assets/1182187/19840455/6eb2582e-9eb3-11e6-92fd-f3551e748f39.png">
<img width="988" alt="screen shot 2016-10-30 at 3 03 31 pm" src="https://cloud.githubusercontent.com/assets/1182187/19840456/6eb2adec-9eb3-11e6-8e5e-8600337747ad.png">
<img width="311" alt="screen shot 2016-10-30 at 3 03 02 pm" src="https://cloud.githubusercontent.com/assets/1182187/19840457/6eb85ada-9eb3-11e6-96b6-22ed798c1e23.png">

**Comment by rawmind0** [comment]
Hi @emcniece ...

thanks for use traefik with rancher, and your feedback...

I think that https://github.com/jmcarbo/rancher-traefik works a little bit different. 

If you run `docker exec -it [contID] curl http://rancher-metadata.rancher.internal` it should work. With alpine 3.4 there is a dns bug that doesn't resolves short names when more than one search domain are in ressolv.conf . It shouldn't affect to confd services, as you could read at the issue https://github.com/rawmind0/alpine-traefik/issues/3

I've tried to connect to your system, but i don't have permission to access the rancher environment that it's empty by default. If you give me access to your rancher environment i could take a look. 

Anyway, if the logs doesn't show anything abnormal, could you review your services traefik labels?? 

Best regards.....

**Comment by emcniece** [comment]
Thanks for the reply. I just tested `docker exec -it [contID] curl http://rancher-metadata.rancher.internal` and confirmed that it works properly from terminal.

While working in terminal on the Rancher node server, I noticed that all of the containers were exited for 7 days. I logged in to the Rancher UI and looked at the Stacks page, and everything was stopped - but then it began initializing and brought all of the containers back online. The weird part is that the Traefik UI now shows the two `whoami` services properly. It just started working and I have no idea why.

I have updated permissions on your account if you want to take a look, but this issue appears to have been magically resolved. Thank you for taking the time to help!
